catroid/src/org/catrobat/catroid/formulaeditor/FormulaElement.java

```diff
@@ -354,6 +354,10 @@ private Object interpretFunction(Functions function, Sprite sprite) {
 				return doubleValueOfLeftChild == null ? 0d : java.lang.Math.toDegrees(Math.atan(doubleValueOfLeftChild));
 			case EXP:
 				return doubleValueOfLeftChild == null ? 0d : java.lang.Math.exp(doubleValueOfLeftChild);
+			case POWER:
+				return ((doubleValueOfLeftChild == null || doubleValueOfRightChild == null)
+						|| (doubleValueOfLeftChild < 0 && doubleValueOfRightChild < 1 && doubleValueOfRightChild > -1))
```

Comment:
There are just too many input/output combinations listed in the [Math.pow](https://developer.android.com/reference/java/lang/Math.html#pow(double, double)) documentation... :smile_cat:
I just tested two cases you catch here (only in java):
```java
Math.pow(-1, 0.5); // == NaN
Math.pow(-1, 0) // == 1.0
```
So, why should both cases return `0` instead?